semeio/jobs/rft/utility.py

```diff
@@ -0,0 +1,265 @@
+            time = datetime.date(year, month, day)
+
+        except ValueError:
+            err_msg = (
+                base_error_msg
+                + "Unable to parse date, expected day month year got: {day} {month} {year}"
+            )
+            raise argparse.ArgumentTypeError(
+                err_msg.format(
+                    line_number=line_number,
+                    filename=filename,
+                    day=tokens[1],
+                    month=tokens[2],
+                    year=tokens[3],
+                )
+            )
+
+        well_times.append((well, time, report_step))
+
+    return well_times
+
+
+def load_and_parse_zonemap_file(filename):
+    """
+    The file must be on the format of <INT STRING*> (any number of strings)
+    which represents <k_value zone_names..>
+
+    Parameters
+    ----------
+    filename : string
+        Filename to open
+
+    Returns
+    -------
+    Dict
+        Returns a dict with k_value as keys and a list of corresponding
+        zonenames
+    """
+
+    zones_at_k_value = {}
+
+    with open(filename, "r") as f:
+        zonemap_lines = f.readlines()
+
+    zonemap_lines = [(strip_comments(l), i + 1) for i, l in enumerate(zonemap_lines)]
+    basic_err_msg = "Line {line_number} in ZoneMap file {filename} not on proper format: 'k zonename <zonename> ...'. "
+    for line, line_number in zonemap_lines:
+        zonemap_line = line.split()
+
+        if not zonemap_line:
+            continue
+
+        if len(zonemap_line) < 2:
+            raise argparse.ArgumentTypeError(
+                basic_err_msg
+                + "Number of zonenames can be 1 or more".format(
+                    line_number=line_number, filename=filename
+                )
+            )
+
+        try:
+            raw_k = int(zonemap_line[0])
+        except ValueError:
+            raise argparse.ArgumentTypeError(
+                basic_err_msg
+                + "k must be integer, was {k}".format(
+                    line_number=line_number, filename=filename, k=zonemap_line[0]
+                )
+            )
+
+        if raw_k == 0:
+            raise argparse.ArgumentTypeError(
+                basic_err_msg
+                + "k values cannot be 0, must start at 1. ".format(
+                    line_number=line_number, filename=filename
+                )
+            )
+
+        k_value = raw_k - 1
+        zones = [zone.strip() for zone in zonemap_line[1:]]
+
+        zones_at_k_value[k_value] = zones
+
+    return zones_at_k_value
+
+
+def valid_eclbase(file_path):
+    """
+    The filename is assumed to be without extension and two files
+    must be present, <filename>.RFT and <filename>.EGRID.
+    Loads both files with respective loaders and returns them
+
+    Parameters
+    ----------
+    filename : string
+        Filename to open
+
+    Returns
+    -------
+    Tuple
+        Returns a tuple with an ecl grid instance and an rft instance
+    """
+    try:
```

Comment:
verify if path exist?

Reply:
The `EclGrid` and `EclRFT` takes a path, it should be their responsibility to check that the path exists (and they do). In case of non-existing path, they throw an `IOError` which we catch.

Reply:
But the error message is hidden. The user does not get the error produced by EclGrid or EclRFTFile. The error message should be propegated to the user.

Reply:
I agree with you. The checking of file also happens further down into the c code in ecl. I've added checks for if the file exists (as this is likely one of the most prominent issues and should preferably have a small error message), and I've included the propagated error message from libecl.

Cheers!